### ISSUE TITLE
orca createSession to return websocket object?

### ISSUE THREAD

**Issue by boynet** [orcajs/reflector#2]
I am kind of noob in this libary is there any chance to return the websocket object to the user so he can send messages to the server? where is the right place to implement it?

**Comment by sharpie7** [comment]
Hi @boynet 

Thanks for raising this comment. The objective in Orca is to make the API independent of the implementation of the signalling so that Orca applications are portable between different signalling systems. As well as this reflector we will have other transport libraries available using a variety of signalling.

Some signalling mechanisms may not use a WebSocket mechanism therefore we cannot, in general, assume that there is always going to be a WebSocket object available in all systems. Even if there is a WebSocket available addressing it directly from the application would expose the application to differnt, propietary, signalling systems which would break the portability.

Can you tell us more about the use-case you want to support? I think there is potentially an important topic here which is to recongize that in the real world applications may not find the Orca API entirely sufficient and may want to access proprietary aspects as well. I am not sure how best to accommodate that.

Regards

Iain

**Comment by boynet** [comment]
I am talking about orcaReflector.js as its clrealy using websocket I want to make some game with room, each room have max of 2 playrs.
the users need to send messages between them and the server. what I did is:(I know its hacky but its the only thing that worked for me)
in orca.js  Session function
````
this.wsconn = function(){
            return sessionImp.wsconn;
        }
````

in my app.js to send message:
````
myUser.session.wsconn().send('some message');
````

and to recive messages:
````
_onmessageParent = myUser.session.wsconn().socket.onmessage;
myUser.session.wsconn().socket.onmessage = function (msg) {
    data =JSON.parse(msg.data);
    if (typeof data === "object" && data.method.what) {
        //my messages always contain what property
    } else {
     ///else call orcaReflector msg handler
        _onmessageParent(msg);
    }
}
````

**Comment by sharpie7** [comment]
Yes - I think the best way to get access to the WS object without changing the API is to add a function to this particular implementation like that you describe. This obviously ties you to the reflector implementation of the Orca API.

We are considering defining extensions to the API to support user messaging. This would mean that applications using Orca could send messages between users without having to reach "inside" the library to see objects that might not exist in other implementations.
